src/compiler/builder.ts

```diff
@@ -451,6 +451,11 @@ function createBuilderProgramState(newProgram: Program, oldState: Readonly<Reusa
                     pendingEmitKind;
             }
         }
+        else if (!canCopySemanticDiagnostics) {
+            // No emit being done, but semantic diagnostics must be recalculated - since these (and the options that change these)
+            // are in the build info file, we need to emit it to keep the info on-disk up-to-date
+            state.buildInfoEmitPending = true;

```

Comment:
@sheetalkamat When I added a test for `noCheck`, it had a `-inconsistencies` baseline for incremental mode, and the culprit seemed to just be that we didn't place an updated `.buildinfo` on disk after the first build that added semantic diagnostics (since we skipped emit, we also skipped updating the build info). This is probably an existing issue with other semantic-diagnostic-affecting-options, too (if they don't also affect emit), but since the fix seemed simple enough, I put it in here, rather than making a separate issue. This change is what's responsible for the handful of `tsbuild` baseline changes in this PR now - they write a `buildinfo` an extra time or two.